### PR TITLE
add docs, widgets, and output schema endpoint for aggregator

### PR DESCRIPTION
**Pull request by albertshau**

### CODE REVIEW

File: core-plugins/docs/GroupByAggregate-batchaggregator.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+Properties
+----------
+**groupByFields:** Comma-separated list of fields to group by.
+Records with the same value for all these fields will be grouped together.
+Records output by this aggregator will contain all the group by fields and aggregate fields.
+For example, if grouping by the ``user`` field and calculating an aggregate ``numActions:count(*)``,
+output records will have a ``user`` field and a ``numActions`` field.
+
+**aggregates:** Aggregates to compute on each group of records.
+Supported aggregate functions are `avg`, `count`, `count(*)`, `first`, `last`, `max`, `min`,`sum`.
```
Comment (awholegunch) [inline core-plugins/docs/GroupByAggregate-batchaggregator.md:23]: What's the difference between `count` and `count(*)`?

File: core-plugins/src/main/java/co/cask/hydrator/plugin/batch/aggregator/GroupByAggregator.java [hunk 2]
```diff
@@ -43,7 +44,7 @@
 @Description("Groups records by a set of fields, then performs aggregates on record fields.")
```
Comment (awholegunch) [inline core-plugins/src/main/java/co/cask/hydrator/plugin/batch/aggregator/GroupByAggregator.java]: Should this not match the description in the first line of the doc description?

File: core-plugins/widgets/GroupByAggregate-batchaggregator.json [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  },
+  "configuration-groups": [
+    {
+      "label": "GroupByAggregate Configuration",
+      "properties": [
+        {
+          "widget-type": "keyvalue",
+          "label": "Aggregates to compute",
+          "name": "aggregates",
+          "description": "List of aggregates to compute for each group. Supported aggregates are count, sum, avg, min, max, first, and last. Each aggregate takes a field as an argument. For example, 'sum(price)' or 'count(*)'",
```
Comment (awholegunch) [inline core-plugins/widgets/GroupByAggregate-batchaggregator.json]: Supported aggregates are avg, count, count(*), first, last, max, min, and sum.

**Comment by sreevatsanraman** [comment]
LGTM

**Pull request merged**